activesupport/lib/active_support/message_verifier.rb

```diff
@@ -44,7 +44,7 @@ def initialize(secret, options = {})
     #   tampered_message = signed_message.chop # editing the message invalidates the signature
     #   verifier.valid_message?(tampered_message) # => false
     def valid_message?(signed_message)
-      return if signed_message.blank?
+      return if signed_message.nil? || !signed_message.valid_encoding? || signed_message.blank?
```

Comment:
Any suggestion? I thought about just using `.to_s` but it seems about `1.4x` slower than checking with `.nil?`